Insights report list rewritten as a table

AllenBW opened the pull request:
fixes https://projects.engineering.redhat.com/browse/RHCLOUD-1186
follows design here: https://redhat.invisionapp.com/share/V4NY3C6XKRW#/screens/356890434

we're gonna save search and filter for phase two of this work 👍 

### ~todo~ nada left to do
* ~proper expand collapse in kebab menu~
* ~card styling still has that wrenched box-shadow~
* ~empty state stuff~
* ~searching~ gonna save this for v2
* ~filtering~ gonna save this for v2 

### full screen screenshot (for those in need of)
<img width="1818" alt="Screen Shot 2019-05-14 at 12 15 08 PM" src="https://user-images.githubusercontent.com/6640236/57714222-0e384a80-7642-11e9-9584-fbb4f3ec85d4.png">
<img width="1820" alt="Screen Shot 2019-05-14 at 12 20 56 PM" src="https://user-images.githubusercontent.com/6640236/57714577-d1208800-7642-11e9-87b3-d39d455bb3e3.png">


### no rule hits, its a good thing
<img width="1560" alt="Screen Shot 2019-05-14 at 11 53 32 AM" src="https://user-images.githubusercontent.com/6640236/57713489-9289ce00-7640-11e9-87c7-0228beb60a3a.png">

### error fetching (absent entity looks the same, different text )
<img width="1339" alt="Screen Shot 2019-05-14 at 11 58 32 AM" src="https://user-images.githubusercontent.com/6640236/57713488-9289ce00-7640-11e9-9c1c-50b17094787d.png">

### our expand and collapse, by default all are expanded

<img width="1500" alt="Screen Shot 2019-05-14 at 12 00 26 PM" src="https://user-images.githubusercontent.com/6640236/57713487-9289ce00-7640-11e9-992e-8cda69aca5b9.png">

### rest of the report looks the same
<img width="1500" alt="Screen Shot 2019-05-14 at 12 00 57 PM" src="https://user-images.githubusercontent.com/6640236/57713485-9289ce00-7640-11e9-964b-0c8395e8e521.png">

AllenBW commented:
cc @katierik  @robcwilliams

Robcwilliams commented:
@AllenBW I like the new layout. A few comments

- If we can add spacing between the icons and the words, they're crunched up together at the moment 

- The ansible remediation iconography / colors confuse me. There is a green checkbox and a black 
checkbox... Not sure what the difference is, but if the black checkbox is meant to say there is no ansible support, this confuses me due to it being a check

AllenBW commented:
@Robcwilliams did the spacing thing! check the second ss for updates

katierik commented:
- There should be nothing when there is no Ansible support (not a black check).
- Ansible text should be the "small" pf text
- Also the labels for "impact","likelihood","total risk" and "risk of change" should be the small type
- Text should be "Ansible remediation" ( not "Ansible Remediation") - Sentence case!
- Why is "ansible remediation" wrapped some places and not others? It should ideally all respond uniformly if it needs to wrap due to screen size.
- Is "Collapse all" and "expand all" in here? Can't tell.
- In the mock there is 16px between the rule title and the impact/likelihood/total risk icons. 

![Xnip2019-05-14_12-24-46](https://user-images.githubusercontent.com/23172038/57714805-49874900-7643-11e9-8bfb-03d320ccaa74.jpg)

![Xnip2019-05-14_12-27-02](https://user-images.githubusercontent.com/23172038/57714929-966b1f80-7643-11e9-9463-589961a47616.jpg)

AllenBW commented:
@katierik  updated screen shot below, changed what we can in the scope of this pr, 
* fixed ansible support size/ case/visibility, thanks for the clarification
* fixed label size
* screen shot below shows the collapse all and expand all (also original post had this)
* added 16px top margin to the battery stack
<img width="1920" alt="Screen Shot 2019-05-14 at 1 13 03 PM" src="https://user-images.githubusercontent.com/6640236/57717813-07153a80-764a-11e9-93d2-c5a035debc2e.png">

katierik commented:
💯 awesome!

The labels with the icons should be middle aligned with each other (wasn't as noticeable before the text size change). Otherwise 👍

AllenBW commented:
That'll have to change over https://github.com/RedHatInsights/insights-frontend-components/blob/master/doc/components/battery.md not something we can so much touch in this work...

AllenBW commented:
@ryelo @karelhala whenever yah all can get a chance,  <3 🙇‍♀ 👍

karelhala commented on src/SmartComponents/Inventory/applications/advisor/Advisor.js:
```diff
@@ -79,84 +94,150 @@ class InventoryRuleList extends Component {
+                    <TableBody/>
+                </Table>
+                :
+                <Card>
```
Is it correct in Card? Other empty states are outside card component.

AllenBW replied:
We need to have the white background and margin around the the message state, i didn't build it into messagestate.js (cuz maybe we wanta abstract it somewhere else n drop it somewhere else 🤷‍♂) by default empty states don't have a background color

ryelo replied:
Was this in the UX mocks? Other places aren't using backgrounds on their empty states. For consistency, this should be revisited.

AllenBW replied:
@katierik can you confirm or deny, white background for empty states?

katierik replied:
Yes white background. [Mock](https://redhat.invisionapp.com/share/V4NY3C6XKRW#/356911950_Empty_State_-_No_Insights). 
The difference between a grey bg and a white bg is this:
- Empty state should be directly on the grey bg if there is a configuration reason it is empty (i.e. it isn't connected yet)
- Empty state should be on the white bg if it is properly configured, but there just isn't anything to show! (e.g. no results after filtering; the system/rule happens to have 0 hits)

AllenBW commented on src/SmartComponents/Inventory/applications/advisor/Advisor.js:
```diff
@@ -79,84 +94,150 @@ class InventoryRuleList extends Component {
+                                <Split>
+                                    <SplitItem> { rule.category.name } &gt; </SplitItem>
+                                    <SplitItem isFilled> { rule.description } </SplitItem>
+                                    <SplitItem style={ { fontSize: global_FontSize_sm.value } }>
```
down with either 🤷‍♂

ryelo replied:
Stylesheets over inline styles every day of the week for me 😄

AllenBW commented on src/SmartComponents/Inventory/applications/advisor/Advisor.js:
```diff
@@ -79,84 +94,150 @@ class InventoryRuleList extends Component {
+                                        }
+                                    </SplitItem>
+                                </Split>
+                                <Section type='icon-group__with-major' style={ { fontSize: global_FontSize_sm.value, marginTop: '16px' } }>
```
I can do this... but for whatever reason.. in dev, i don't see these styles applied when they are done via a class
<img width="1580" alt="Screen Shot 2019-05-15 at 7 46 19 AM" src="https://user-images.githubusercontent.com/6640236/57773256-97548d80-76e5-11e9-9cb1-ad4445273e26.png">

(see the highlighted row below)

AllenBW replied:
but i do see them when applied directly as styles to elements

ryelo replied:
Something might not be scoped right or the specificity isn't correct. This page is using other styles from `advisor.scss`.

karelhala commented:
@AllenBW thanks for the screenshots! Can you remove the Remediate with ansible and kebab button if there are no rules?

AllenBW commented:
@karelhala YOUR WISH IS MY COMMAND!! 
<img width="1328" alt="Screen Shot 2019-05-15 at 10 39 55 AM" src="https://user-images.githubusercontent.com/6640236/57784487-f6be9780-76fd-11e9-98de-6d6dae8bc0e6.png">

katierik commented:
Not a blocker I don't think, but we should still have a toolbar visible on the page even when empty. Will be more relevant when we have filter etc on there. [Mock](https://redhat.invisionapp.com/share/V4NY3C6XKRW#/356911950_Empty_State_-_No_Insights) has it. 
Its part of the empty state pattern.

AllenBW commented:
@katierik and here's our failed state
<img width="1247" alt="Screen Shot 2019-05-15 at 10 51 52 AM" src="https://user-images.githubusercontent.com/6640236/57785318-844eb700-76ff-11e9-8493-d44945c33620.png">

nlcwong commented:
@AllenBW hey Allen, when we get the "failed state" what http error code is the UI getting?

AllenBW commented:
@nlcwong anything that is not a `20x` 🤔 

@katierik with love from keyboard
<img width="1443" alt="Screen Shot 2019-05-15 at 12 25 13 PM" src="https://user-images.githubusercontent.com/6640236/57792398-e82bac80-770c-11e9-90be-3a4af46937cf.png">

karelhala commented:
:tada: This PR is included in version 3.41.66 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@red-hat-insights/insights-frontend-components)
- [GitHub release](https://github.com/RedHatInsights/insights-frontend-components/releases/tag/v3.41.66)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
